### PR TITLE
Add build options for emacs depends on #2386

### PR DESCRIPTION
**Pull request by healther**
Update of #3102 
I somehow screwed up my pushes, and this was the simplest solution.

Edit:
We currently maintain two repositories for our local spack installation, one with things that at least someone at our group thought could be useful upstream. I will ask why this change was necessary, but don't know that yet.

depends on #2386

### CODE REVIEW

File: var/spack/repos/builtin/packages/emacs/package.py [hunk 1]
```diff
@@ -60,6 +60,14 @@ def configure_args(self):
...
-            args = ['--without-x']
+            args = ['--without-x',
+                    '--with-jpeg=no',
+                    '--with-gif=no',
+                    '--with-tiff=no',
+                    '--with-xpm=no',
+                    '--with-x-toolkit=no',
+                    ]
+            if '+gtkplus' in spec:
+                raise InstallError('The variant +gtkplus is ignored if ~X is selected. Aborting.')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/emacs/package.py]: The flake8 tests are failing because this line is too long.

**Comment by citibeth** [comment]
I'm wondering if this needs a little more revision:

1. The `X` variant is pretty useless, I would get rid of it.

2. I would suggest that we rename `toolkit` to `gui`, and standardize on that variant name where appropriate.

3. @alalazo's work on enum-valued variants will allow removal of the checking of the `toolkit/gui` variant value.

**Comment by citibeth** [comment]
What is the impetus behind this PR?

**Comment by healther** [comment]
The further options were needed to make it build without having ```X``` installed and if ```X``` isn't installed the other packages are likely not needed either.

**Comment by citibeth** [comment]
I would revise to something like this (note, this requires #2386):
```

class Emacs(AutotoolsPackage):
    """The Emacs programmable text editor."""

    homepage = "https://www.gnu.org/software/emacs"
    url      = "http://ftp.gnu.org/gnu/emacs/emacs-24.5.tar.gz"

    version('25.1', '95c12e6a9afdf0dcbdd7d2efa26ca42c')
    version('24.5', 'd74b597503a68105e61b5b9f6d065b44')

    # Once 2386 is merged
    #variant('gui', default=None,
    #        description='GUI toolkit to use (requires X11)',
    #        values=('gtk', 'athena'))

    # No type-checking for now...
    variant('gui', default=None)

    depends_on('ncurses')
    depends_on('libtiff', when='+X')
    depends_on('libpng', when='+X')
    depends_on('libxpm', when='+X')
    depends_on('giflib', when='+X')
    depends_on('libx11', when='+X')
    depends_on('libxaw', when='+X toolkit=athena')
    depends_on('gtkplus+X', when='+X toolkit=gtk')

    def configure_args(self):
        spec = self.spec
        toolkit = spec.variants['toolkit'].value
        if self.variants['gui'].value is None:
            args = ['--without-x',
                    '--with-jpeg=no',
                    '--with-gif=no',
                    '--with-tiff=no',
                    '--with-xpm=no',
                    '--with-x-toolkit=no',
                    ]
        else:
            args = [
                '--with-x',
                '--with-x-toolkit={0}'.format(spec.variants['gui].value)
            ]

    return args
```

**Comment by citibeth** [comment]
I would revise to something like this (note, this requires #2386):

@alalazo How do we cahnge `when='+X'` to the equivalent of "when variants['gui'] is not None"?
```

class Emacs(AutotoolsPackage):
    """The Emacs programmable text editor."""

    homepage = "https://www.gnu.org/software/emacs"
    url      = "http://ftp.gnu.org/gnu/emacs/emacs-24.5.tar.gz"

    version('25.1', '95c12e6a9afdf0dcbdd7d2efa26ca42c')
    version('24.5', 'd74b597503a68105e61b5b9f6d065b44')

    # Once 2386 is merged
    #variant('gui', default=None,
    #        description='GUI toolkit to use (requires X11)',
    #        values=('gtk', 'athena'))

    # No type-checking for now...
    variant('gui', default=None)

    depends_on('ncurses')
    depends_on('libtiff', when='+X')
    depends_on('libpng', when='+X')
    depends_on('libxpm', when='+X')
    depends_on('giflib', when='+X')
    depends_on('libx11', when='+X')
    depends_on('libxaw', when='gui=athena')
    depends_on('gtkplus+X', when='gui=gtk')

    def configure_args(self):
        spec = self.spec
        toolkit = spec.variants['toolkit'].value
        if self.variants['gui'].value is None:
            args = ['--without-x',
                    '--with-jpeg=no',
                    '--with-gif=no',
                    '--with-tiff=no',
                    '--with-xpm=no',
                    '--with-x-toolkit=no',
                    ]
        else:
            args = [
                '--with-x',
                '--with-x-toolkit={0}'.format(spec.variants['gui].value)
            ]

    return args
```

**Comment by healther** [comment]
Can someone put a label ```depends on 2386``` on this PR? Or tell me how to do it?

**Comment by davydden** [comment]
@healther mention https://github.com/LLNL/spack/pull/2386 in the OP?

**Pull request closed without merging**